File: samples/python3/sample.py
# Copyright (c) 2024 デジタル庁
# 
# This software is released under the MIT License.
# https://opensource.org/licenses/MIT

import requests

version = "v1"
method = "simple" # or detail
kanji="日本［東京］　花子"
kana="トウキョウ　ハナコ"
key="REPLACE_WITH_YOUR_API_KEY"

url = f"https://api.kktg.digital.go.jp/{version}/{method}"

params = {
  'kanji': kanji,
  'kana': kana,
  'key': key,
}

response = requests.get(url, params=params)

# レスポンスのステータスコードを確認
print(response.status_code)  # 200

# レスポンスのコンテンツを表示
print(response.text)

File: samples/nodejs/sample.js
// Copyright (c) 2025 デジタル庁
// 
// This software is released under the MIT License.
// https://opensource.org/licenses/MIT

import https from 'https';

(async () => {

  const method = 'simple'; // or detail
  const version = 'v1';

  const url = new URL(`https://api.trueno-kktg.digital.go.jp/${version}/${method}`);
  url.search = new URLSearchParams({
    'kanji': '日本　太郎',
    'kana': 'にほん たろう',
    'key': '(api key)',
  });

  const response = await (new Promise((resolve) => {
    https.get(url, (response) => {
      const buffer = [];
      response.on('data', chunk => buffer.push(chunk));
      response.once('end', () => {
        resolve({
          statusCode: response.statusCode,
          body: buffer.join(''),
        })
      })
    });
  }));

  // レスポンスの表示
  console.log(response);

})();